### DISCUSSION TITLE
Limit of ChatAI usage.

### DISCUSSION THREAD

**Discussion by CarlosNahuelcoy** [Luke100000/minecraft-comes-alive#662] [Q&A]
Hello, I just discovered this mod and I am completely amazed by all the options it offers. I was particularly impressed by the option to chat with villagers using AI. I understand that there is a limitation where they forget the conversation to some extent, but are there any other limitations? Like limits per hour, per day, or something like that? I would really appreciate an answer so that I don't abuse that option and exhaust it without realizing it.

**Comment by Luke100000** [comment]
After currently 45 minutes they forget what you said before exactly. There is a hourly token limit to protect my wallet ^^ Longer dialogues (with more active memory needed to be process) require more token. Other than that there is no limit.
A bit more infos are here: https://github.com/Luke100000/minecraft-comes-alive/wiki/GPT3-based-conversations

**Comment by CarlosNahuelcoy** [comment]
Oh, wow, thank you so much for your very honest response. It makes me very happy to know that there isn't such a strict limit that would prevent me from advancing my game as I chat with the villagers. Another little question if it's not too much trouble, below the option to activate the AI there was another one called "villagerChatAIServer". Does that influence anything? Is there anything that can be customized there? Thank you so much for creating such a cool mod.
![Screenshot_3](https://github.com/Luke100000/minecraft-comes-alive/assets/89953468/9ac22393-cd12-4b7b-9ccd-92d563fe8765)

**Comment by Luke100000** [comment]
This is the server address which handles AI related stuff, theoretically you could host your own one, or in my case I use a testing server.

**Reply by Explosif3005** [reply]
What do that mean? Do all the AI discusion pass by said server to be handled?

**Reply by Luke100000** [reply]
yes
